rpcs3/Emu/Cell/Modules/sceNp.cpp

```diff
@@ -3960,49 +4034,120 @@ error_code sceNpScoreGetBoardInfo(s32 transId, SceNpScoreBoardId boardId, vm::pt
 		return SCE_NP_COMMUNITY_ERROR_INVALID_ONLINE_ID;
 	}
 
+	auto trans_ctx = idm::get<score_transaction_ctx>(transId);
+
+	if (!trans_ctx)
+	{
+		return SCE_NP_COMMUNITY_ERROR_INVALID_ID;
+	}
+
+	nph.get_board_infos(trans_ctx, boardId, boardInfo, async);
+
+	if (async)
+	{
+		return CELL_OK;
+	}
+
+	return *trans_ctx->result;
+}
+
+error_code sceNpScoreGetBoardInfo(s32 transId, SceNpScoreBoardId boardId, vm::ptr<SceNpScoreBoardInfo> boardInfo, vm::ptr<void> option)
+{
+	sceNp.warning("sceNpScoreGetBoardInfo(transId=%d, boardId=%d, boardInfo=*0x%x, option=*0x%x)", transId, boardId, boardInfo, option);
+
+	if (auto res = scenp_score_get_board_info(transId, boardId, boardInfo, option, false); res)
+	{
+		return res;
+	}
```

Comment:
this if logic feels obsolete if CELL_OK is returned anyway if it succeeds...
Same holds at many other places.

Reply:
It has been discussed on Discord.
In summary:
-if I return an error_code it will result in a extra print of the error code
-if I just return the function return value the log will show Failed with 0